README.md

```diff
@@ -15,73 +15,90 @@ Due to the fact that hardly any information can be encoded in `Type`, we rely on
 
-   otherwise.
- * If groups of a given type are known to be finite, one needs to define
-   > `Base.length(G::Group) = order(Int, G)`
 
-   which is a "best effort", cheap computation of length of the group iterator. For practical reasons the largest group you could possibly iterate over is of order ~`factorial(19)` (which still fits into `Int`).
+Additionally the following assumptions are placed on the iteration:
+ * `first(G)` must be the identity
+ * iteration over an infinite group should exhaust every fixed radius ball (in word-length metric) around the identity in finite time.
 
-##### Obligatory methods
+#### Obligatory methods
  * `Base.one(G::Group)`: return the identity of the group
- * `GroupsCore.order(::Type{I}, G::Group) where I<:Integer`: the order of `G` returned as an instance of `I`; only arbitrary size integers are required to return mathematically correct answer.
+ * `GroupsCore.order(::Type{I}, G::Group) where I<:Integer`: the order of `G` returned as an instance of `I`; only arbitrary size integers are required to return mathematically correct answer. An infinite group must throw `GroupsCore.InfiniteOrder` exception.
  * `GroupsCore.gens(G::Group)`: return a random-accessed collection of generators of `G`; if a group does not come with a generating set (or it may be prohibitively expensive to compute), one needs to alter `GroupsCore.hasgens(::Group) = false`.
  * `Base.rand(rng::Random.AbstractRNG, rs::Random.Sampler{GT}) where GT<:Group`: to enable asking for random group elements treating group as a collection, i.e. calling `rand(G, 2, 2)`.
 
 ## `GroupElement` methods
-##### Obligatory methods
- * `Base.parent(g::GroupElement)`: return the parent object of a given group element. Parent objects of the elements of the same group must be **identical** (i.e. ===).
+#### Obligatory methods
+ * `Base.parent(g::GroupElement)`: return the parent object of a given group element. Parent objects of the elements of the same group must be **identical** (i.e. `===`).
  * `GroupsCore.parent_type(::Type{<:GroupElement})`: given the type of an element return the type of its parent.
- * `GroupsCore.istrulyequal(g::GEl, h::GEl) where GEl<:GroupElement`: return the mathematical equality of group elements; by default the standard equality `==` calls this function.
- * `GroupsCore.hasorder(g::GroupElement)`: return `true` if `g` has finite order (without computing it)`.
+ * `GroupsCore.==(g::GEl, h::GEl) where GEl<:GroupElement`: return the mathematical equality of group elements;
+ * `GroupsCore.isfiniteorder(g::GroupElement)`: return `true` if `g` has finite order (possibly without computing it)`. If `isfiniteorder(g)` returns `false`, `order(g)` is required to throw `GroupsCore.InfiniteOrder` exception.
```

Comment:
Is that what we settled on? I mean, surely `order(BigInt, g)` or `order(fmpz, g)` etc. should do that; but wasn't the option of using something like https://gist.github.com/kalmarek/640630a53d5837a5c7fe401eb2f047bd for `order` also still on the table, and/or returning something like a `Union{BigInt, SomeTypeRepresentingInfiniteOrder}` still an option?

Reply:
SomeTypeRepresentingInfiniteOrder could be a bunch of things: a Float type and we use `inf`; our own type representing infinity; we could also (ab)use `Nothing`  ("this group/element has no finite order, so we can return nothing")

Reply:
We left it to better times: see the discussion in #8